sqlglot/dataframe/sql/dataframe.py

```diff
@@ -365,20 +365,27 @@ def select(self, *cols, **kwargs) -> DataFrame:
                     x.this for x in get_tables_from_expression_with_join(self.expression)
                 ]
                 cte_names_in_join = [x.this for x in join_table_identifiers]
+                # If we a columns resolves to multiple CTE expressions then we want to use each CTE left-to-right
+                # and therefore we allow multiple columns with the same name in the result. This matches the behavior
+                # of Spark.
+                resolved_column_position: t.Dict[str, int] = {}
                 for ambiguous_col in ambiguous_cols:
                     ctes_with_column = [
                         cte
                         for cte in self.expression.ctes
                         if cte.alias_or_name in cte_names_in_join
                         and ambiguous_col.alias_or_name in cte.this.named_selects
                     ]
-                    # If the select column does not specify a table and there is a join
-                    # then we assume they are referring to the left table
-                    if len(ctes_with_column) > 1:
-                        table_identifier = self.expression.args["from"].args["expressions"][0].this
+                    if ambiguous_col in resolved_column_position:
+                        cte = seq_get(ctes_with_column, resolved_column_position[ambiguous_col] + 1)
```

Comment:
i don't understand this code, can you explain it?

Reply:
Simplified a bit and left a comment